api/internal/target/kusttarget.go

```diff
@@ -332,16 +334,29 @@ func (kt *KustTarget) removeValidatedByLabel(rm resmap.ResMap) error {
 // accumulateResources fills the given resourceAccumulator
 // with resources read from the given list of paths.
 func (kt *KustTarget) accumulateResources(
-	ra *accumulator.ResAccumulator, paths []string) (*accumulator.ResAccumulator, error) {
+	ra *accumulator.ResAccumulator, paths []string, origin *resource.Origin) (*accumulator.ResAccumulator, error) {
 	for _, path := range paths {
 		// try loading resource as file then as base (directory or git repository)
-		if errF := kt.accumulateFile(ra, path); errF != nil {
+		if errF := kt.accumulateFile(ra, path, origin); errF != nil {
 			ldr, err := kt.ldr.New(path)
 			if err != nil {
 				return nil, errors.Wrapf(
 					err, "accumulation err='%s'", errF.Error())
 			}
-			ra, err = kt.accumulateDirectory(ra, ldr, false)
+			originCopy := origin.Copy()
+			repoSpec, err := git.NewRepoSpecFromUrl(path)
+			if err == nil {
+				origin.Repo = repoSpec.Host + repoSpec.OrgRepo
+				absPath := repoSpec.AbsPath()
+				path = absPath[strings.Index(absPath[1:], "/")+1:][1:]
+				origin.Path = ""
+				origin.Ref = repoSpec.Ref
+			}
+			origin.Path = filepath.Join(origin.Path, path)
+			ra, err = kt.accumulateDirectory(ra, ldr, origin, false)
+			// reset the origin so that it does not retain path information from
+			// the directory accumulation
+			origin = &originCopy
```

Comment:
the above new lines should be in a method on `Origin` (perhaps `func Append(path) *Origin`) with regression coverage in origin_test

Reply:
done